### PR TITLE
D3ASIM-2893: Added restore simulation while the simulation is launched.

### PR DESCRIPTION
**Pull request by spyrostz**

### CODE REVIEW

File: src/d3a/d3a_core/cli.py [hunk 1]
```diff
@@ -19,8 +19,7 @@
 from logging import getLogger
 
 import click
-import dill
```
Comment (hannesdiedrich) [inline src/d3a/d3a_core/cli.py:22]: Now that you removed all usages of `dill` you could remove it from the requirements.

File: src/d3a/d3a_core/simulation.py [hunk 2]
```diff
@@ -448,15 +423,14 @@ def _handle_input(self, console, sleep: float = 0):
...
+                if cmd not in {'i', 'p', 'q', 'r', 'R', 's', '+', '-'}:
                     log.critical("Invalid command. Valid commands:\n"
                                  "  [i] info\n"
                                  "  [p] pause\n"
                                  "  [q] quit\n"
                                  "  [r] reset\n"
-                                 "  [S] stop\n"
+                                 "  [s] stop\n"
                                  "  [R] start REPL\n"
-                                 "  [s] save state\n"
```
Comment (hannesdiedrich) [inline src/d3a/d3a_core/simulation.py:459]: Thanks a lot for this refactoring! I guess that it is not worth it to reuse the new current_state into a pickle file for this option? WE never used it right? So I agree to remove it.

**Review by hannesdiedrich — APPROVED** [review]
Thanks for the additional refactoring! LGTM

File: src/d3a/d3a_core/simulation.py (hunk 2 above)
Comment (spyrostz) [reply src/d3a/d3a_core/simulation.py:459]: No worries, we never used pickle for saving the state of the simulation (it used to be part of our legacy code, was not working right from the start IIRC), and it is guaranteed to fail (Area and Market classes are not pickleable)

File: src/d3a/d3a_core/cli.py (hunk 1 above)
Comment (spyrostz) [reply src/d3a/d3a_core/cli.py:22]: Will do so , thanks !

File: src/d3a/d3a_core/simulation.py [hunk 3]
```diff
@@ -241,24 +222,18 @@ def deactivate_areas(self, area):
...
-                slot_resume, tick_resume = divmod(self.area.current_tick,
-                                                  self.simulation_config.ticks_per_slot)
-            else:
+            if initial_slot == 0:
                 self.run_start = DateTime.now(tz=TIME_ZONE)
                 self.paused_time = 0
-                slot_resume = tick_resume = 0
+                slot_resume = 0
+            else:
+                slot_resume = initial_slot
```
Comment (faizan2590) [inline src/d3a/d3a_core/simulation.py:234]: I think `slot_resume ` could be pulled out of this if else block, simply via always setting `slot_resume = initial_slot`

**Review by faizan2590 — APPROVED** [review]
Left one comment, other than that LGTM

File: src/d3a/d3a_core/simulation.py (hunk 3 above)
Comment (spyrostz) [reply src/d3a/d3a_core/simulation.py:234]: Agreed, fixing this

**Review by faizan2590 — APPROVED** [review]
lgtm

**Pull request merged**